compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/transform-fire/multiple-scope.js

```diff
@@ -0,0 +1,21 @@
+// @enableFire
+import {fire} from 'react';
+
+function Component(props) {
+  const foo = props => {
+    console.log(props);
+  };
+  useEffect(() => {
+    fire(foo(props));
+    function innerNested() {
+      fire(foo(props));
+      function nested() {
+        fire(foo(props));
+      }
+    }
+
+    nested();
```

Comment:
`nested` isn't defined since in the scope of innerNested

Reply:
d'oh!